packages/flutter_tools/lib/src/run_cold.dart

```diff
@@ -72,7 +72,10 @@ class ColdRunner extends ResidentRunner {
     // Connect to observatory.
     if (debuggingOptions.debuggingEnabled) {
       try {
-        await connectToServiceProtocol();
+        await Future.wait(<Future<void>>[
+          connectToServiceProtocol(),
+          serveDevToolsGracefully(),
+        ]);
```

Comment:
`attach` wasn't being called from `flutter run` for profile mode like it was for debug mode, so I had to add this to `run` in addition to `attach` for the cold runner

Reply:
What happens when we attach to a running profile mode app?

Reply:
attach is called in run_cold.dart